src/lib/Parser.hs

```diff
@@ -784,13 +863,21 @@ inpostfix' p op = Postfix $ do
 mkName :: String -> Name
 mkName s = Name SourceName (fromString s) 0
 
+nameToStr :: Name -> String
+nameToStr = tagToStr . nameTag
+
+mkInterfaceConsName :: Name -> Name
+mkInterfaceConsName =
+  GlobalName . fromString . (++ " _cons") . nameToStr
```

Comment:
Ok, this will have to go. We should never have automatically generated global names, because they can enter into conflict with user defined names

Reply:
Based on what we discussed offline, I just replace my " _cons" (which did contain a space, making it in principle an invalid user-defined identifier?) with "#", which serves the same purpose and is more readable anyway! Thanks!